barcode/templates/barcode/baseextra.html

```diff
@@ -0,0 +1,98 @@
+{% extends "baseextra.html" %}
+{{ super() }}
+
+{% if detect_barcode_scan %}
+  <script type="text/javascript">
+      var getBadgeNumFromBarcode = function (barcode, callback) {
+          barcode = ('' + barcode).trim();
+          callback = callback || function () {
+              };
+
+          var regexpUUID = /^[0-9a-f]{8}-?[0-9a-f]{4}-?[1-5][0-9a-f]{3}-?[89ab][0-9a-f]{3}-?[0-9a-f]{12}$/i;
```

Comment:
Registration uses check-in barcodes, which return UUIDs when scanned, so we also want to avoid attemtping to decrypt anything that's a UUID. Regex taken from: https://stackoverflow.com/a/29742838